packages/web-client/app/models/workflow/workflow-card.ts

```diff
@@ -52,6 +52,8 @@ export class WorkflowCard extends WorkflowPostable {
   }
 
   @action async onComplete() {
+    // not too sure if this is the right fix, but it seems to make sense to not be able to complete a completed card
+    if (this.isComplete) return;
```

Comment:
Do you know of a case where this happens?

Reply:
layer two connect card is the place, hence the comment in the same commit

Reply:
Comment removed as it would be confusing out of context - this referenced the layer two connect card calling `onComplete` in `next`, leading to it starting test waiters after a restored workflow had finished rendering.